jobbergate-api/jobbergate_api/cli/dev_server.py

```diff
@@ -0,0 +1,42 @@
+from time import sleep
+
+import typer
+import uvicorn
+from loguru import logger
+from sqlalchemy import create_engine
+
+from jobbergate_api.config import settings
+
+
+def _wait_for_db(wait_count, wait_interval):
+    count = 0
+    while count < wait_count:
+        logger.debug(f"Checking health of database at {settings.DATABASE_URL}: Attempt #{count}")
+        count += 1
+        try:
+            engine = create_engine(settings.DATABASE_URL)
+            with engine.connect() as db:
+                db.execute("select version()")
+            return
+        except Exception as err:
+            logger.warning(f"Database is not yet healthy: {err}")
+        sleep(wait_interval)
+
+    raise RuntimeError("Could not connect to the database")
+
+
+def dev_server(
+    db_wait_count: int = typer.Option(3, help="How many times to attempt a check"),
+    db_wait_interval: float = typer.Option(5.0, help="Seconds to wait between checks"),
+    port: int = typer.Option(5000, help="The port where the server should listen"),
+    log_level: str = typer.Option("DEBUG", help="The level to log uvicorn output"),
+):
+    try:
+        _wait_for_db(db_wait_count, db_wait_interval)
+    except Exception:
+        logger.error("Database is not available")
+        typer.Exit(1)
```

Comment:
a good docker-compose.yaml file wouldn't require this. Take the [armada API example](https://github.com/omnivector-solutions/armada-api/blob/main/docker-compose.yaml#L26-L38)

Reply:
We have a docker-compose.yaml that also spins up db services if you prefer to use that approach.